core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -368,6 +352,100 @@ public void write(MarshalledEntry<? extends K, ? extends V> marshalledEntry) {
       }
    }
 
+   private ByteBuffer createByteBuffer(FileEntry entry) {
+      MarshalledEntry marshalledEntry = entry.marshalledEntry;
+      org.infinispan.commons.io.ByteBuffer key = marshalledEntry.getKeyBytes();
+      org.infinispan.commons.io.ByteBuffer data = marshalledEntry.getValueBytes();
+      org.infinispan.commons.io.ByteBuffer meta = marshalledEntry.getMetadataBytes();
+
+      ByteBuffer buf = ByteBuffer.allocate(entry.size);
+      buf.putInt(entry.size);
+      buf.putInt(entry.keyLen);
+      buf.putInt(entry.dataLen);
+      buf.putInt(entry.metadataLen);
+      buf.putLong(entry.expiryTime);
+      buf.put(key.getBuf(), key.getOffset(), key.getLength());
+      buf.put(data.getBuf(), data.getOffset(), data.getLength());
+      if (meta != null)
+         buf.put(meta.getBuf(), meta.getOffset(), meta.getLength());
+      buf.flip();
+      return buf;
+   }
+
+   private FileEntry allocateAndCreateFileEntry(MarshalledEntry entry,
+                                                org.infinispan.commons.io.ByteBuffer key,
+                                                org.infinispan.commons.io.ByteBuffer data,
+                                                org.infinispan.commons.io.ByteBuffer meta) {
+      int metadataLength = meta == null ? 0 : meta.getLength();
+      int len = KEY_POS + key.getLength() + data.getLength() + metadataLength;
+      FileEntry newEntry = allocate(len);
+      long expiryTime = meta != null ? entry.getMetadata().expiryTime() : -1;
+      return new FileEntry(newEntry, key.getLength(), data.getLength(), metadataLength, expiryTime, entry);
+   }
+
+   private FileEntry allocateAndCreateFileEntry(MarshalledEntry entry) {
+      org.infinispan.commons.io.ByteBuffer key = entry.getKeyBytes();
+      org.infinispan.commons.io.ByteBuffer data = entry.getValueBytes();
+      org.infinispan.commons.io.ByteBuffer meta = entry.getMetadataBytes();
+
+      return allocateAndCreateFileEntry(entry, key, data, meta);
+   }
+
+   @Override
+   public void writeBatch(Collection<MarshalledEntry<? extends K, ? extends V>> marshalledEntries) {
+      resizeLock.readLock().lock();
+      try {
+         List<FileEntry> fileEntries = marshalledEntries.stream().map(this::allocateAndCreateFileEntry)
+               .collect(Collectors.toList());
```

Comment:
The sort is the same reason as process, assuming it is faster to write to disk sequentially to prevent random locations on disk. Although the more I think about it, writes may not have this issue since it should be sequential anyways unless the size is small enough to fill a gap. Wonder if it is worth it here.